### ISSUE TITLE
Wrong results for Dynamic Data > User / Author

### ISSUE THREAD

**Issue by maltmann-muc** [soflyy/oxygen-bugs-and-features#2019]
WordPress 3.7
Oxygen 3.7.1

**1) User Display Name**
If inserting Dynamic Data > User > Display Name the result is always the nice name of the user, not the name configured in user profile. 
(Works fine for Author > Display Name)

**2) User > Meta Field**
if inserting Dynamic Data > User > Meta Field the result is an Array instead of a string.
That's because of a wrong attribute name is passed / evaluated.
Passed: $atts['key']
Evaluated: $atts['meta_key']
See screenshot.
<img width="1487" alt="Bildschirmfoto 2021-03-23 um 08 51 12" src="https://user-images.githubusercontent.com/51136650/112113031-967cc480-8bb6-11eb-87dd-c88a1a6ac5aa.png">

**3) Dynamic Data > Author > Meta Field** 
e.g. 'first_name'/'last_name' also doesn't work. Returns configured user's display name instead.

**Comment by shoelaced** [comment]
+1 -- I'm also running into the issue of the user display name being incorrect. It only shows the username.

**Comment by gabrieluntura** [comment]
+1

**Comment by McVoy** [comment]
Hi, 
On some forum I read that someone (Niko) posted code here that was temporarily solving the problem. Do any of you have any temporary solution? 
Thank you in advance.

**Comment by adrien-robert** [comment]
+1

**Comment by eduardoyi** [comment]
+1

**Comment by samjco** [comment]
[oxygen data='user_meta' key='last_name']
[oxygen data='user_meta' key='first_name']
All does not work :(

**Comment by yewzy** [comment]
+1

**Comment by guyinpv** [comment]
The fix is a tiny little patch and I'm shocked this has been sitting here for a year and a half with broken meta field lookups. Just, why?

```
PHP Warning:  Undefined array key "meta_key" in /home/......./plugins/oxygen/component-framework/includes/oxygen-dynamic-shortcodes.php on line 417
```

The original source:

```
function oxygen_user_meta($atts) {
  $userdata = $this->oxygen_get_userdata($atts['id']);
  return get_user_meta($userdata->ID, $atts['meta_key'], true);
}
```

How this function is used, for the user meta key lookup, well there is no 'meta_key' field in `$atts`.
I'm seeing it as just 'key'.

My hotfix is to directly edit the code and change 'meta_key' to 'key' if needed.

```
if (array_key_exists('meta_key', $atts)) {
  return get_user_meta($userdata->ID, $atts['meta_key'], true);
} else {
  return get_user_meta($userdata->ID, $atts['key'], true);
}
```

Then it works for any meta key lookups.

Obviously this could be overwritten the next time Oxygen updates, so it's no permanent fix.

I could have just changed 'meta_key' to 'key' without checking if the array key exists, but I don't know if this function is used within any other context where it actually should be 'meta_key', so just to be safe I wanted both options.

I'm not using this patch in production though, I had to write my own PHP function and call that instead for getting meta data.

Hopefully the devs can fix this quickly, seems like such a small fix.

**Comment by Spellhammer** [comment]
Fixed in 4.2 beta 1.
